data/transactions/logic/TEAL_opcodes.md

```diff
@@ -727,3 +741,114 @@ params: account index, asset id. Return: did_exist flag (1 if exist and 0 otherw
 
 
 params: txn.ForeignAssets offset. Return: did_exist flag (1 if exist and 0 otherwise), value.
+
+## assert
+
+- Opcode: 0x72
+- Pops: *... stack*, uint64
+- Pushes: _None_
+- immediately fail unless value X is a non-zero number
+- LogicSigVersion >= 3
+
+## min_balance
+
+- Opcode: 0x73
+- Pops: *... stack*, uint64
+- Pushes: uint64
+- get minimum balance for the requested account specified by Txn.Accounts[A] in microalgos. A is specified as an account index in the Accounts field of the ApplicationCall transaction, zero index means the sender
+- LogicSigVersion >= 3
+- Mode: Application
+
+## getbit
+
+- Opcode: 0x74
+- Pops: *... stack*, {any A}, {uint64 B}
+- Pushes: uint64
+- pop a target A (integer or byte-array), and index B. Pushes the Bth bit of A.
+- LogicSigVersion >= 3
+
+bit indexing begins with low-order bits in integers. Bit 4 is the 1 in int 0x0010. Indexing begins in the first bytes of a byte-string (as seen in getbyte and substring). Bits 0 through 11 are 1 in byte 0xfff000000000
+
+## setbit
+
+- Opcode: 0x75
+- Pops: *... stack*, {any A}, {uint64 B}, {uint64 C}
+- Pushes: uint64
+- pop a target A, index B, and bit C. Sets the Bth bit of A to C, and push the result
+- LogicSigVersion >= 3
+
+see explanation of bit ordering in getbit
+
+## getbyte
+
+- Opcode: 0x76
+- Pops: *... stack*, {[]byte A}, {uint64 B}
+- Pushes: uint64
+- pop a byte-array A and integer B. Extract the Bth byte of A and push it as an integer
+- LogicSigVersion >= 3
+
+## setbyte
+
+- Opcode: 0x77
+- Pops: *... stack*, {[]byte A}, {uint64 B}, {uint64 C}
+- Pushes: []byte
+- pop a byte-array A, integer B, and small integer C (between 0..255). Set the Bth byte of A to C, and push the result
+- LogicSigVersion >= 3
+
+## swap
+
+- Opcode: 0x78
+- Pops: *... stack*, {any A}, {any B}
+- Pushes: *... stack*, any, any
+- swaps two last values on stack: A, B -> B, A
+- LogicSigVersion >= 3
+
+## select
+
+- Opcode: 0x79
+- Pops: *... stack*, {any A}, {any B}, {uint64 C}
+- Pushes: any
+- selects one of two values to retain: A, B, C -> A ? B : C
+- LogicSigVersion >= 3
+
+## dig
+
+- Opcode: 0x80 {uint8 depth}
+- Pops: *... stack*, any
+- Pushes: *... stack*, any, any
+- duplicate value N from the top of stack
+- LogicSigVersion >= 3
+
+## stxn
```

Comment:
That's true.  I meant it as "stack txn" or "slot txn", but we do use stxn for signed all over the place.  What do you think of the "ge" in getxn as meaning? group extended?

I considered adding an i, for indirect: gtxni/gtxnia but the trouble there is that "i" can also mean "immediate" which describes the normal gtxn exactly.  So "i" doesn't help much.

With substring, we have substring3, which I guess means that it takes 3 args on stack, unlike the normal one which only takes 1.  So I guess that would mean gtxn1/gtxna1 for this opcode.  But that's ugly, and gtxn2 isn't much better (and less principled).

What about atxn/atxna for "alternate"? It also it nice because we normally call the top-of-stack "A" when describing an opcode that takes one thing from the stack, so an alternate way to think of "atxn" is that it's txn that reads the index from "A".